src/ripple/rpc/Version.h

```diff
@@ -0,0 +1,46 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLED_RIPPLE_RPC_VERSIONS_H
+#define RIPPLED_RIPPLE_RPC_VERSIONS_H
+
+#include <beast/module/core/diagnostic/SemanticVersion.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/rpc/impl/JsonObject.h>
+
+namespace ripple {
+namespace RPC {
+
+enum class Version {first, recommended, last};
+
+beast::SemanticVersion const& version (Version);
+
+template <class Object>
+void setVersion(Object& parent)
+{
+    auto&& object = addObject (parent, jss::version);
```

Comment:
I have a lot to learn on the `auto&&` front.  The way I read this, `object` may be an rvalue reference.  Under what situation would that be desirable?  Might it be better to have a plain `auto&` and have the compiler bark if `addObject` returned an rvalue?